docs/technical-details/reference/components/configuration.md

```diff
@@ -0,0 +1,85 @@
+# Component Configuration
+
+> For information on the distinction between applications and extensions, refer to beginning of the [Applications](../../../developers/applications) documentation section.
+
+Harper components are configured with a `config.yaml` file located in the root of the component module directory. This file is how an components configures other components it depends on. Each entry in the file starts with a component name, and then configuration values are indented below it.
+
+```yaml
+name:
```

Comment:
Is there a general `config.yml` schema that we could show here? I think it would be helpful if we said here's the blueprint for a component config file. Maybe something like this?

```yaml
rest: true/false
extension: "path/to/extension.js"
"<component_name>":
	package: "<npm_package>"
	files: "<pattern>"
```

Reply:
I think the difference in configurations between like root config, extensions/plugins, and applications is enough to just keep this generic. I'm trying to essentially explain yaml without explaining yaml haha. The point of this is that each "entry" is tab-delimited. Other parts of the docs actually dig into the specifics of what goes in to the file.